### PR TITLE
VPC-SC factory JSON Schemas

### PR DESCRIPTION
**Pull request by ludoo**
As discussed, this only covers the VPC-SC module and does not yet integrate the schemas in FAST.

### CODE REVIEW

File: modules/vpc-sc/schemas/ingress-policy.schema.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "VPC-SC ingress policy",
+  "type": "object",
+  "required": [
+    "from",
+    "to"
+  ],
+  "properties": {
+    "from": {
+      "type": "object",
```
Comment (wiktorn) [inline modules/vpc-sc/schemas/ingress-policy.schema.json:11]:
```suggestion
      "type": "object",
      "additionalProperties": false,
      "required": [
        "access_levels"
      ],
```
Additional properties to catch if someone misspells the attribute.

Required - I vaguely remember that not providing access level bite me some time ago with not working policy. Encoding that in schema will be of a great help.

**Review by wiktorn — COMMENTED** [review]
Generally - :heart: :heart: :heart: - as this may help me a lot in the future and I was bitten by this beast already.

One comment regarding making the schemas a bit more specific - set "additionalProperties" to false on all objects in JSON schema.

The other improvement I see, is to mark more fields as required based on experience (but this may easily explode into big schema, if we need to use "anyOf" or similar approaches).

File: modules/vpc-sc/schemas/ingress-policy.schema.json (hunk 1 above)
Comment (ludoo) [reply modules/vpc-sc/schemas/ingress-policy.schema.json:11]: for ingress policies, access levels are not required: if you specify resources (i.e. projects) then you don't need access levels; if you don't have resources, then you need either a star of a list of access levels

TL;DR variable type is correct

**Comment by ludoo** [comment]
> Generally - ❤️ ❤️ ❤️ - as this may help me a lot in the future and I was bitten by this beast already.
> 
> One comment regarding making the schemas a bit more specific - set "additionalProperties" to false on all objects in JSON schema.
> 
> The other improvement I see, is to mark more fields as required based on experience (but this may easily explode into big schema, if we need to use "anyOf" or similar approaches).

schema needs to be a 1:1 translation of the variable type :)

File: modules/vpc-sc/schemas/ingress-policy.schema.json (hunk 1 above)
Comment (wiktorn) [reply modules/vpc-sc/schemas/ingress-policy.schema.json:12]: I'm not saying it's not correct, I'm just saying it could be better, but IIUC, this means going into `anyOf`, where one requires `access_levels`  and the other that requires `resources`, so then we encode that one of them is provided.

But it will be PITA to encode those rules in JSON Schema, goes outside what we can express in HCL, so I would just keep in mind of this limitation.
